TeamCityQueueScraper.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Prometheus.Client;
+using TeamCitySharp;
+
+namespace TeamCityBuildStatsScraper
+{
+    internal class TeamCityQueueScraper: IHostedService, IDisposable
+    {
+        private readonly IHost _host;
+        private readonly IConfiguration _configuration;
+        private Timer _timer;
+
+        public TeamCityQueueScraper(IHost host, IConfiguration configuration)
+        {
+            _host = host;
+            _configuration = configuration;
+        }
+        
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // Fire off the Scraper starting *right now* and do it again every minute
+            _timer = new Timer(ScrapeQueueStats, null, TimeSpan.Zero, TimeSpan.FromMinutes(1));
+
+            return Task.CompletedTask;
+        }
+
+        private void ScrapeQueueStats(object state)
+        {
+            var metricFactory = _host.Services.GetRequiredService<IMetricFactory>();
+            var teamCityToken = _configuration.GetValue<string>("TEAMCITY_TOKEN");
+            var teamCityUrl = _configuration.GetValue<string>("BUILD_SERVER_URL");
+            var teamCityClient = new TeamCityClient(teamCityUrl, true);
+
+            teamCityClient.ConnectWithAccessToken(teamCityToken);
```

Comment:
Can these be done in the ctor instead?

Reply:
🤔 wouldn't this try to keep using the same session for the life of the service pod, though? I agree it'd make the method less heavy but as I understand it putting that connection in the class ctor would mean it's opened once at startup and never closed, whereas this way I'm pretty sure that it's opened and disposed every time it's called by the `Timer`.

Let's revisit this when you're back.